pgcli/pgcompleter.py

```diff
@@ -300,21 +307,18 @@ def _match(item):
                     # fuzzy matches
                     return -float('Infinity'), -match_point
 
-        # Fallback to meta param if meta_collection param is None
-        meta_collection = meta_collection or repeat(meta)
-        # Fallback to 0 if priority_collection param is None
-        priority_collection = priority_collection or repeat(0)
-
-        collection = zip(collection, meta_collection, priority_collection)
-
         matches = []
+        for cand in collection:
+            if isinstance(cand, _Candidate):
+                item, prio, display_meta, synonyms = cand
+            else:
+                item, display_meta, prio, synonyms = cand, meta, 0, [cand]
 
-        for item, meta, prio in collection:
-            sort_key = _match(item)
+            sort_key = max(_match(x) for x in synonyms)
```

Comment:
I think the `sort_key` assigment could be moved into the `if isinstance` block to avoid a lot of unnecessary list allocations:

```
if isinstance(cand, _Candidate):
    item, prio, display_meta, synonyms = cand
    sort_key = max(_match(x) for x in synonyms)
else:
    item, display_meta, prio = cand, meta, 0
    sort_key = _match(cand)
````